### PR TITLE
Document execution dev

### PR DESCRIPTION
**Pull request by BojanSovticEngIT**
Note: On my env, we changed so that the domain is always the same (localhost).

TODO:
- Metadata service (service needs to be changed for saving files)
- Notes changes (service needs to be changed for saving multiple notes)
- Send mail service is bugged on my env

### CODE REVIEW

File: knowage-vue/src/helpers/commons/fileHelper.ts [hunk 1]
```diff
@@ -29,6 +29,7 @@ export function downloadDirectFromResponseWithCustomName(response, customFilenam
 }
 
 export function downloadDirect(jsonData, filename, contentType) {
+	console.log("CONTENT TYPE: ", contentType)
```
Comment (Redjaw) [inline knowage-vue/src/helpers/commons/fileHelper.ts:32]: Note for me to remove this after merge.

File: knowage-vue/src/modules/documentExecution/dossier/Dossier.routes.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 let routes = [
     {
-        path: '/dossier/:id',
+        path: '/dossier-temp/:id',
```
Comment (Redjaw) [inline knowage-vue/src/modules/documentExecution/dossier/Dossier.routes.js:3]: Is this route needed or just a test typo?

File: knowage-vue/src/modules/documentExecution/main/dialogs/documentExecutionLinkDialog/DocumentExecutionLinkDialog.vue [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+            this.getPublicUrl()
+        },
+        loadDocument() {
+            this.document = this.propDocument
+        },
+        loadParameters() {
+            this.linkParameters = this.parameters as any[]
+        },
+        getPublicUrl() {
+            const tenet = (this.$store.state as any).user.organization
```
Comment (Redjaw) [inline knowage-vue/src/modules/documentExecution/main/dialogs/documentExecutionLinkDialog/DocumentExecutionLinkDialog.vue:85]: Note for me to rename it to tenant after merge

File: knowage-vue/src/modules/documentExecution/main/dialogs/documentExecutionMetadataDialog/DocumentExecutionMetadataDialog.vue [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+            this.$emit('close')
+        },
+        save() {
+            this.$emit('saveMetadata', this.metadata)
+        }
+    }
+})
+</script>
+
+<style lang="scss">
```
Comment (Redjaw) [inline knowage-vue/src/modules/documentExecution/main/dialogs/documentExecutionMetadataDialog/DocumentExecutionMetadataDialog.vue:183]: put scoped please if not needed for children, the pi-upload class is too generic

**Review by Redjaw — CHANGES_REQUESTED** [review]
A few components (execution and sidenav) are very verbose, is it possibile to split part of them?

File: knowage-vue/src/modules/documentExecution/dossier/Dossier.routes.js (hunk 2 above)
Comment (BojanSovticEngIT) [reply knowage-vue/src/modules/documentExecution/dossier/Dossier.routes.js:3]: It was conflicting with new routes, i guess it is not needed anymore, should I remove this file completely?

Comment (BojanSovticEngIT) [reply knowage-vue/src/modules/documentExecution/dossier/Dossier.routes.js:3]: Execution and parameter sidenav have a lot of logic/formatting inside them, should I put something in separate files? @Redjaw

Comment (Redjaw) [reply knowage-vue/src/modules/documentExecution/dossier/Dossier.routes.js:3]: > It was conflicting with new routes, i guess it is not needed anymore, should I remove this file completely?

yes, thx

Comment (Redjaw) [reply knowage-vue/src/modules/documentExecution/dossier/Dossier.routes.js:3]: > Execution and parameter sidenav have a lot of logic/formatting inside them, should I put something in separate files? @Redjaw

yes, but only If it is possible without complicating too much or breaking the structure. Thx

File: knowage-vue/src/modules/documentExecution/main/dialogs/documentExecutionMetadataDialog/DocumentExecutionMetadataDialog.vue (hunk 4 above)
Comment (Redjaw) [reply knowage-vue/src/modules/documentExecution/main/dialogs/documentExecutionMetadataDialog/DocumentExecutionMetadataDialog.vue:183]: Ok, leave it, I will take care of the class to add.

**Review by Redjaw — APPROVED** [review]

**Pull request merged**